### ISSUE TITLE
IDISP004 on Dispatcher.Invoke

### ISSUE THREAD

**Issue by jeremyVignelles** [DotNetAnalyzers/IDisposableAnalyzers#194]
I'm not really sure about what to do with this one, but I found something that was perfectly fine before, and that now triggers a warning:

In WPF, calling the following code
```cs
await System.Windows.Threading.Dispatcher.CurrentDispatcher.Invoke(() => Task.FromResult(42)).ConfigureAwait(false);
```
results in `IDISP004 - Don't ignore created IDisposable.`

Which disposable is it talking about?

Please also note that `Task.CompletedTask` doesn't trigger any warning, nor is it triggering any warning when storing the result in a variable:

```cs
var r = await System.Windows.Threading.Dispatcher.CurrentDispatcher.Invoke(() => Task.FromResult(42)).ConfigureAwait(false);
```

I'm upgrading from 2.1.2 (VS2017) to 3.1.0 (VS2019)

Edit : Funny note: `var _ =` does trigger the warning

**Comment by JohanLarsson** [comment]
Thanks for reporting, regression.

**Comment by JohanLarsson** [comment]
`Task` implements  `IDisposable` https://referencesource.microsoft.com/#mscorlib/system/threading/Tasks/Task.cs,045a746eb48cbaa9
But we specialcase it to not nag.
